orion/pkg/controllers/class_controller_test.go

```diff
@@ -0,0 +1,445 @@
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program1_2020_spring_class1", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var class domains.Class
+	if err := json.Unmarshal(recorder.Body.Bytes(), &class); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+	assert.EqualValues(t, "program1", class.ProgramId)
+	assert.EqualValues(t, "2020_spring", class.SemesterId)
+	assert.EqualValues(t, "class1", class.ClassKey)
+	assert.EqualValues(t, "program1_2020_spring_class1", class.ClassId)
+	assert.EqualValues(t, "churchill", class.LocationId)
+	assert.EqualValues(t, "3 pm - 5 pm", class.Times)
+	assert.EqualValues(t, now, class.StartDate)
+	assert.EqualValues(t, later, class.EndDate)
+}
+
+func TestGetClass_Failure(t *testing.T) {
+	classService.mockGetByClassId = func(classId string) (domains.Class, error) {
+		return domains.Class{}, errors.New("not found")
+	}
+	services.ClassService = &classService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/class/program2_2020_summer", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusNotFound, recorder.Code)
+}
+
+//
+// Test Get Classes by other properties
+//
+func TestGetClassesByProgram_Success(t *testing.T) {
+	classService.mockGetByProgramId = func(programId string) ([]domains.Class, error) {
+		return createMockClasses(), nil
+	}
+	services.ClassService = &classService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/classes/program/program1", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var classes []domains.Class
+	if err := json.Unmarshal(recorder.Body.Bytes(), &classes); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+
+	assertMockClasses(t, 1, classes[0])
+	assertMockClasses(t, 2, classes[1])
+	assertMockClasses(t, 3, classes[2])
+	assert.EqualValues(t, 3, len(classes))
+}
+
+func TestGetClassesBySemester_Success(t *testing.T) {
+	classService.mockGetBySemesterId = func(semesterId string) ([]domains.Class, error) {
+		return createMockClasses(), nil
+	}
+	services.ClassService = &classService
+
+	// Create new HTTP request to endpoint
+	recorder := sendHttpRequest(t, http.MethodGet, "/api/classes/v1/classes/semester/2020_summer", nil)
+
+	// Validate results
+	assert.EqualValues(t, http.StatusOK, recorder.Code)
+	var classes []domains.Class
+	if err := json.Unmarshal(recorder.Body.Bytes(), &classes); err != nil {
+		t.Errorf("unexpected error: %v\n", err)
+	}
+
+	assertMockClasses(t, 3, classes[0])
+	assertMockClasses(t, 4, classes[1])
+	assert.EqualValues(t, 2, len(classes))
+}
+
+func TestGetClassesByProgramAndSemester_Success(t *testing.T) {
+	classService.mockGetByProgramAndSemesterId = func(programId, semesterId string) ([]domains.Class, error) {
+		return createMockClasses(), nil
```

Comment:
I think this is the culprit to your tests failing. You are mocking the service to return ALL mocked classes. So the list of "found" programs and semester becomes the entire mocked class list. Consider filtering out classes here.